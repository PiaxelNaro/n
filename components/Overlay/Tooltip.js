// @flow

import React from 'react';
import styled from 'styled-components';
import Overlay from '.';

const ContentWrapper = styled.span`
    border-bottom: 1px dotted #aaa;
    cursor: help;
`;

const TooltipWrapper = styled.div`
    color: #999;
    padding: 8px 6px;
    font-size: 12px;
    font-weight: normal;
    line-height: 1.5;
`;

const Tooltip = ({
    content,
    children,
    showArrow = true,
}: {
    content: any,
    children: any,
    showArrow?: boolean,
}) =>
    <Overlay.Trigger
        trigger="hover"
        getOverlay={() =>
            <Overlay showArrow={showArrow}>
                <TooltipWrapper>{content}</TooltipWrapper>
            </Overlay>}
    >
        <ContentWrapper>{children}</ContentWrapper>
    </Overlay.Trigger>;

export default Tooltip;
