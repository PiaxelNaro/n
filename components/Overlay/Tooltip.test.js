// @flow

import React from 'react';
import { shallow } from 'enzyme';
import Tooltip from './Tooltip';

describe('<Tooltip />', () => {
    it('renders text and tooltip', () => {
        const tooltip = shallow(<Tooltip content="Lorem ipsum dolor.">Lorem</Tooltip>);
        expect(tooltip).toMatchSnapshot();
        expect(
            shallow(tooltip.find('OverlayTrigger').prop('getOverlay')())
        ).toMatchSnapshot();
    });
});
