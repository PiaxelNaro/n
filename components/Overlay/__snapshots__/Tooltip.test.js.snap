// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> renders text and tooltip 1`] = `
<OverlayTrigger
  getOverlay={[Function]}
  trigger="hover"
>
  <Tooltip__ContentWrapper>
    Lorem
  </Tooltip__ContentWrapper>
</OverlayTrigger>
`;

exports[`<Tooltip /> renders text and tooltip 2`] = `
<Overlay__OverlayWrapper
  showArrow={true}
>
  <Overlay__ArrowUp />
  <Overlay__OverlayInner>
    <Tooltip__TooltipWrapper>
      Lorem ipsum dolor.
    </Tooltip__TooltipWrapper>
  </Overlay__OverlayInner>
</Overlay__OverlayWrapper>
`;
